src/ahrens_lab_to_nwb/yu_mu_cell_2019/yu_mu_cell_2019_segmentation_extractor.py

```diff
@@ -0,0 +1,82 @@
+"""Modification of the roiextractors.Hdf5ImagingExtractor to read the custom miroscope data from the Ahrens lab."""
+from pathlib import Path
+from typing import List, Optional
+
+import h5py
+import numpy as np
+from numpy.typing import ArrayLike
+from roiextractors.segmentationextractor import SegmentationExtractor
+from neuroconv.utils import FilePathType
+
+
+class YuMu2019SegmentationExtractor(SegmentationExtractor):
+    """Custom extractor for reading segmentation data for the Yu Mu 2019 Cell paper."""
+
+    extractor_name = "YuMu2019SegmentationExtractor"
+    mode = "file"
+
+    def __init__(self, file_path: FilePathType, sampling_frequency: float):
+        super().__init__()
+        self._kwargs = dict(file_path=str(Path(file_path).absolute()), sampling_frequency=sampling_frequency)
+        self._sampling_frequency = sampling_frequency
+        self.file_path = file_path
+        self._file = h5py.File(name=file_path)
+
+        baseline_shape = self._file["baseline"].shape
+        assert baseline_shape != (1, 6,), (  # The MATLAB half-precision coding stores the dataset as a (1,6) object
+            "The flourescence series in this file are saved in half precision - file cannot be read in Python! "
+            "Please recast the MATLAB datatype and save a new file."
+        )
+
+        self._roi_response_raw = self._file["baseline"]
+        self._roi_response_dff = self._file["timeseries"]  # TODO: Notes describe this as 'detrended'
+        self._image_masks = 1
```

Comment:
Is this necessary?

Reply:
It must have been but I don't recall why. Probably should instead be something fixed with ROIExtractors, will try to recall the issue

Reply:
Removed in #11